src/update-manager.js

```diff
@@ -660,11 +670,21 @@ class UpdateManager {
       })
 
       var protocol = this.editable(doc.uri, kb)
-      if (!protocol) {
-        throw new Error("Can't make changes in uneditable " + doc)
+      if (protocol === false) {
+        throw new Error("Update: Can't make changes in uneditable " + doc)
       }
-      var i
-      if (protocol.indexOf('SPARQL') >= 0) {
+      if (protocol === undefined) { // Not enough metadata
+        if (secondTry) {
+          throw new Error("Update: Loaded " + doc + "but stil can't figure out what editing protcol it supports.")
+        }
+        console.log(`Update: have not loaded ${doc} before: loading now...`)
+        this.store.fetcher.load(doc).then( response => {
+          this.update(deletions, insertions, callback, true) // secondTry
```

Comment:
So we are just guessing the update protocol here then?

Reply:
No, we are doing a GET to get get the HTTP headers in between, so the second try will normally work as it will know what protocol to use.

Reply:
This is atomic to the extent that we need it, I think -- that the thing either succeeds or fails.  If it fails 409 that means the thing you wanted to delete wasn't there.   So you can use things to make a locl, or a semaphore, or a P/V flag as they used to be called.
The server has to implement the atomicity.